File: benchmark/results/python-invocation-script/run-002/b-baseline/output/metaengine_demo/shipping/services/carrier_gateway.py
"""CarrierGateway service."""
from typing import Optional

from metaengine_demo.shipping.aggregates.shipment import Shipment


class CarrierGateway:
    """CarrierGateway service for the shipping domain."""

    def create(self, input: Optional[Shipment]) -> Shipment:
        raise NotImplementedError("not implemented")

    def find_by_id(self, id: str) -> Optional[Shipment]:
        raise NotImplementedError("not implemented")

    def list(self, limit: float) -> list[Shipment]:
        raise NotImplementedError("not implemented")

    def delete(self, id: str) -> None:
        raise NotImplementedError("not implemented")

File: benchmark/results/ts-monolith-sonnet-inline/run-005/b-baseline/output/src/orders/fulfillment/services/CarrierGateway.ts
import { Id } from '../../../shared/value-objects/Id';
import { Shipment } from '../aggregates/Shipment';

/** Gateway for integrating with external carrier APIs. */
export class CarrierGateway {
  create(input: Partial<Shipment>): Shipment { throw new Error('not implemented'); }
  findById(id: Id): Shipment | null { throw new Error('not implemented'); }
  list(limit: number): Shipment[] { throw new Error('not implemented'); }
  delete(id: Id): void { throw new Error('not implemented'); }
}
